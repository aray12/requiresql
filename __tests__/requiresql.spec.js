'use strict';

const requireSQL = require('../index.js');

describe('requiresql', () => {
  it('should load the sql into JS objects', () => {
    const queries = requireSQL(__dirname, '/queries.sql');

    expect(queries).toMatchSnapshot();
  })
})
