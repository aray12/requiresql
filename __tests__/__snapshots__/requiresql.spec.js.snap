// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requiresql should load the sql into JS objects 1`] = `
Object {
  "complexSelect": "SELECT
  g.group_id,
  g.group_type_code,
  g.group_name,
  g.is_inactive,
  COALESCE(g_count.invited_count, 0) AS invited_count,
  COALESCE(g_count.following_count, 0) AS following_count,
  m.message_body,
  CASE WHEN g.sponsor_percent is NULL THEN 0 ELSE g.sponsor_percent END AS sponsor_percent,
  CASE WHEN g.sponsor_monthly_limit is NULL THEN 0 ELSE g.sponsor_monthly_limit END AS sponsor_monthly_limit,
  CASE WHEN g.brand_percent is NULL THEN 0 ELSE g.brand_percent END AS brand_percent,
  a.currency_base AS currency_group
FROM \\"group\\" AS g
LEFT OUTER JOIN account AS a ON g.account_id = a.account_id
LEFT OUTER JOIN message AS m ON g.message_id = m.message_id
LEFT OUTER JOIN (
  SELECT
    group_id,
    sum(CASE WHEN ga_count.is_following = TRUE AND ga_count.is_invited = TRUE THEN 1 END) AS following_count,
    sum(CASE WHEN ga_count.is_invited = TRUE THEN 1 END) AS invited_count
  FROM group_account AS ga_count
  GROUP BY ga_count.group_id
) AS g_count ON (g_count.group_id = g.group_id)
WHERE g.account_id = ?
ORDER BY g.group_name ASC;
",
  "dropTable": "DROP TABLE testing CASCADE;
",
  "selectAll": "SELECT * FROM my_table;
",
}
`;
